sources/engine/Stride.BepuPhysics/Stride.BepuPhysics.Navigation/Components/RecastNavigationComponent.cs

```diff
@@ -152,31 +157,8 @@ private void Rotate()
         {
             return;
         }
-        var position = Entity.Transform.WorldMatrix.TranslationVector;
 
-        float angle = (float)Math.Atan2(Path[0].Z - position.Z,
-            Path[0].X - position.X);
-
-        Entity.Transform.Rotation = Quaternion.RotationY(-angle);
+        var position = Entity.Transform.WorldMatrix.TranslationVector;
+        Entity.Transform.Rotation = Quaternion.LookRotation(Vector3.Normalize(Path[0] - position), Vector3.UnitY);
```

Comment:
This method could be merged in with `Move`, that way you can use the `WorldToLocal` extension mentioned above to transform both positon and rotation at the same time